src/client/datascience/jupyter/kernelVariables.ts

```diff
@@ -421,6 +422,37 @@ export class KernelVariables implements IJupyterVariables {
         return result;
     }
 
+    public async getVariableProperties(
```

Comment:
The alternative is to fetch these attributes on the original IJupyterVariable, which might help with returning a hover more quickly.

Reply:
I have another idea for this. Improve IPython such that it returns these items for tensors in the inspect request. Then we use that to get the values (this way jupyter gets improved tensor support too).

We could do that in parallel to this though.

Reply:
Good idea. Would we file an issue for them and offer to submit a PR? Or just submit the PR?

Reply:
File an issue, then submit a PR I'd say.

Reply:
Opened an issue here: https://github.com/ipython/ipython/issues/12816